src/libclang/libclang_parser.cpp

```diff
@@ -373,6 +398,36 @@ void libclang_compile_config::do_set_flags(cpp_standard standard, compile_flags
         else
             add_flag("-std=c++1z");
         break;
+    case cpp_standard::cpp_17:
+        if (detail::libclang_compile_config_access::clang_major_version(CPPAST_CLANG_BINARY) >= 5) {
+            if (flags & compile_flag::gnu_extensions)
+                add_flag("-std=gnu++17");
+            else
+                add_flag("-std=c++17");
+            break;
+        }
+        else
+            throw std::invalid_argument("c++17 is not yet supported for current version of clang");
+    case cpp_standard::cpp_2A:
+        if (detail::libclang_compile_config_access::clang_major_version(CPPAST_CLANG_BINARY) >= 9) {
+            if (flags & compile_flag::gnu_extensions)
+                add_flag("-std=gnu++2A");
+            else
+                add_flag("-std=c++2A");
+            break;
+        }
+        else
+            throw std::invalid_argument("c++2A is not yet supported for current version of clang");
+    case cpp_standard::cpp_20:
+        if (detail::libclang_compile_config_access::clang_major_version(CPPAST_CLANG_BINARY) >= 10) {
```

Comment:
Correct me if I'm wrong but it is clang version 10 that supports C++20?
And clang version 9 for C++2A
And clang version 5 for C++17